deno2/main.rs

```diff
@@ -0,0 +1,39 @@
+extern crate libc;
+use std::ffi::CStr;
+use std::ffi::CString;
+use libc::c_char;
+use libc::c_int;
+
+#[link(name = "deno", kind = "static")]
+extern "C" {
+    fn deno_v8_version() -> *const c_char;
+    fn deno_init();
+    fn deno_set_flags(argc: *mut c_int, argv: *const *const c_char);
+}
+
+fn set_flags() {
+    // Create a vector of zero terminated strings.
+    let args = std::env::args()
+        .map(|arg| CString::new(arg).unwrap())
+        .collect::<Vec<CString>>();
```

Comment:
I believe `Vec<_>` would be more idiomatic.

Reply:
Would it even need the Vec<>?
That would seem to be implied

Reply:
I don't think so. `Vec` does not differ from any other `FromIterator` implementation in this context.

Reply:
True actually there isn't enough context available however as args only seems to be used in creating argv you should be able to simplify

```rust
let args = std::env::args()
    .map(|arg| CString::new(arg).unwrap())
    .collect::<Vec<CString>>();
let argv = args.iter()
    .map(|arg| arg.as_ptr())
    .collect::<Vec<*const c_char>>();
```
To
```rust
let argv = std::env::args()
    .map(|arg| CString::new(arg).unwrap().as_ptr())
    .collect::<Vec<_>>();
```

Reply:
Thanks! Done.

Reply:
@ZanderBrown, that's not correct, actually. If you did the same thing with Rust references instead of raw pointers, the compiler would reject it immediately. Raw pointers, on the other hand, are just `usize` numbers without any memory safety semantics, so the result of `CString::as_ptr` is copied onto the closure's stack, then the string is dropped and the closure returns a dangling pointer. The separate `args` vector keeps the strings alive until it is dropped itself.

@piscisaureus PTAL